lib/ca_certs.ml

```diff
@@ -99,12 +102,25 @@ let trust_anchors () =
         | "OpenBSD" -> detect_one openbsd_location
         | "Linux" -> detect_list linux_locations
         | "Darwin" ->
-            let cmd =
-              Bos.Cmd.(
-                v "security" % "find-certificate" % "-a" % "-p"
-                % macos_keychain_location)
-            in
-            Bos.OS.Cmd.(run_out cmd |> out_string |> success)
+            macos_keychain_locations
+            |> List.map (fun path ->
+                   let cmd =
+                     Bos.Cmd.(
+                       v "security" % "find-certificate" % "-a" % "-p" % path)
+                   in
+                   Bos.OS.Cmd.(
+                     run_out cmd |> out_string |> success |> Result.to_option))
+            |> List.fold_left
+                 (fun acc cert ->
+                   match (cert, acc) with
+                   | Some cert, Some acc -> Some (cert ^ "\n" ^ acc)
+                   | Some cert, None -> Some cert
+                   | _ -> acc)
```

Comment:
instead of `option`, if we use `result` here, we can remove the `Option.to_result` below, and also retain some error message.